rpc/auth.go

```diff
@@ -200,6 +200,60 @@ func MakeSimpleMultiAuthHandler(forEntities, expectedPayloads []string) AuthHand
 	})
 }
 
+// MakeSimpleMultiAuthPairHandler works similarly to MakeSimpleMultiAuthHandler with the addition of
+// supporting a key, id pair used to ensure that a key that maps to the id matches the key passed
+// during the function call.
+func MakeSimpleMultiAuthPairHandler(forEntities, expectedPayloads []map[string]string) AuthHandler {
+	if len(expectedPayloads) == 0 {
+		panic("expected at least one payload")
+	}
+
+	// Create a list of KeyIDs for the entities from our given map since MakeEntitiesChecker takes in a list of entities
+	var forEntitiesList []string
+	for _, entity := range forEntities {
+		for key := range entity {
+			forEntitiesList = append(forEntitiesList, key)
+		}
+	}
+
+	entityChecker := MakeEntitiesChecker(forEntitiesList)
+	return AuthHandlerFunc(func(ctx context.Context, entity, payload string) (map[string]string, error) {
```

Comment:
[q] what is the advantage of having this as an inline function?

Reply:
Ah I see this is done above. Uh. Ok I have no idea why this is; seems to obfuscate the code.

Reply:
`MakeSimpleMultiAuthPairHandler` is basically defining this as the callback function right?

Reply:
@ohEmily no real advantage that I see here. Was following the convention in the `MakeSimpleMultiAuthHandler` that is similar to `MakeSimpleMultiAuthPairHandler`

Reply:
ah fwiw I figured this out. It's because the return type of the method is actually a function; we are returning the callback rather than calling the function.

Reply:
^ thats what i was trying to get at above. shouldve explained that better

Reply:
I should've also explained it better